extensions/ql-vscode/src/qlpack-generator.ts

```diff
@@ -0,0 +1,85 @@
+import { mkdir, writeFile } from "fs-extra";
+import { dump } from "js-yaml";
+import { join } from "path";
+import { CodeQLCliServer } from "./cli";
+
+export type QueryLanguage =
```

Comment:
We already have multiple places where languages are hardcoded. It would be nice not to add another one. (Even better would be to consolidate them.)

I found 4(-ish) locations:

- https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/src/helpers.ts#L533
- https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/src/packaging.ts#L11
- https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/src/remote-queries/gh-api/variant-analysis.ts#L13
- https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/src/remote-queries/shared/variant-analysis.ts#L30

Reply:
Agreed. Arguably in this case you don't need the enum. We should already have the logic to infer the language name from the database, and then we pass that same inferred language name to `packAdd` without checking it again.

Reply:
@aeisenberg Thanks for rounding up all the places where we repeat the language type. My intention is to open a follow-up PR to consolidate all these places. 👍 

>  We should already have the logic to infer the language name from the database

I was thinking the same, but actually the database allows any string to be set as a language as we don't do validation on what we store on that field (as far as I can tell). So you could end up generating a QL pack for an unknown/unsupported language. 

But I agree we could have one place to store info on the language. 👍

Reply:
The database contains a `*.dbscheme` file, and the VS Code extension infers the language name from the dbscheme file name (which follows certain naming conventions). Such a file can only exist if there was a valid CodeQL extractor to produce the database. So it is IMO valid to assume that the inferred language name is valid, without doing further enum-based validation.

In the rare case of working with a DB that was produced by an experimental extractor with no published library pack (i.e. someone has locally created the extractor and DB but there is no `codeql/<lang>-all` pack published yet), generating a starter CodeQL query pack using your new command will fail, which is perfectly acceptable.

Reply:
> My intention is to open a follow-up PR to consolidate all these places. 

Sounds good. 👍